### ISSUE TITLE
Cutted graphs 

### ISSUE THREAD

**Issue by clebig** [ntop/ntopng#2168]
My traffic/app graphs are displayed with regular cuts (every 5 minutes).

I observed my grafana graphs are also affected the same way.

Also, every 5 minutes in influxd logs, I can see a 400 http error code.

`Nov 20 15:00:01 gatewayNG influxd[15788]: [httpd] ::1 - - [20/Nov/2018:15:00:01 +0100] "POST /write?db=ntopng HTTP/1.1" 204 0 "-" "-" 92919e6c-eccc-11e8-9a3d-000000000000 24224

Nov 20 15:00:13 gatewayNG influxd[15788]: [httpd] ::1 - - [20/Nov/2018:15:00:13 +0100] "POST /write?db=ntopng HTTP/1.1" 400 91 "-" "-" 99b79269-eccc-11e8-9a3e-000000000000 173703

Nov 20 15:00:25 gatewayNG influxd[15788]: [httpd] ::1 - - [20/Nov/2018:15:00:25 +0100] "POST /write?db=ntopng HTTP/1.1" 204 0 "-" "-" a0ddc83f-eccc-11e8-9a3f-000000000000 29187
![screenshot_20181120_150227](https://user-images.githubusercontent.com/17143757/48778573-e9d46480-ecd5-11e8-802a-c2ec22d477a0.png)
![screenshot_20181120_090551](https://user-images.githubusercontent.com/17143757/48778574-e9d46480-ecd5-11e8-911c-2ae8ce7a25da.png)
![screenshot_20181120_090605](https://user-images.githubusercontent.com/17143757/48778575-e9d46480-ecd5-11e8-9320-a73050ae7743.png)
![screenshot_20181120_092254](https://user-images.githubusercontent.com/17143757/48778576-ea6cfb00-ecd5-11e8-9a9e-80f104c86b12.png)
![screenshot_20181120_150256](https://user-images.githubusercontent.com/17143757/48778577-ea6cfb00-ecd5-11e8-91bd-88c592eb8e99.png)

`

**Comment by lucaderi** [comment]
Please also see #2154 as the graph issue described in this bug might be related to it

**Comment by simonemainardi** [comment]
do you see any error in the ntopng logs?

`journalctl -u ntopng`

please, also visit alerts page and see if there are any anomalies

![image](https://user-images.githubusercontent.com/3397663/48787986-28741a00-ecea-11e8-8ed9-5ea328befbbf.png)

**Comment by simonemainardi** [comment]
to better debug we are going to
- raise alerts when influxdb POST fail
- preserve the latest ~5 files with datapoints to troubleshoot

**Comment by clebig** [comment]
Please find attached the journalctl ntopng log.

I have no abnormal warning or error in "Detected Alerts", just alerts about a few flooder hosts.

[journalctl_-u.log](https://github.com/ntop/ntopng/files/2601691/journalctl_-u.log)

**Comment by clebig** [comment]
> to better debug we are going to
> 
>     * raise alerts when influxdb POST fail
> 
>     * preserve the latest ~5 files with datapoints to troubleshoot

Is this addressed to me ? If so, how do I do that ?

**Comment by simonemainardi** [comment]
they are just notes for us. @emanuele-f will help you.

**Comment by emanuele-f** [comment]
@clebig please check out #2097 to fix the `max-series-per-database limit exceeded` error. Then please get back to us and report the new log

**Comment by clebig** [comment]
> @clebig please check out #2097 to fix the `max-series-per-database limit exceeded` error. Then please get back to us and report the new log

Ok, I've just added INFLUXDB_DATA_MAX_SERIES_PER_DATABASE=1500000 in /etc/default/influxdb and restarted influxdb. 
I can still see 400 codes. 
`Nov 22 10:30:22 gatewayNG influxd[19316]: [httpd] ::1 - - [22/Nov/2018:10:30:22 +0100] "POST /write?db=ntopng HTTP/1.1" 204 0 "-" "-" 3c349c46-ee39-11e8-8008-000000000000 26778`

I'll keep looking at the graphs to see if I get new cuts.

**Comment by simonemainardi** [comment]
actually 204 is a 'no content' error. @emanuele-f do you think there are cases where we can send empty files to ntopng?

**Comment by clebig** [comment]
![screenshot from 2018-11-22 10-52-14](https://user-images.githubusercontent.com/17143757/48895358-28d4f800-ee45-11e8-8f74-cda86b8ad8de.png)
![screenshot from 2018-11-22 10-53-30](https://user-images.githubusercontent.com/17143757/48895359-28d4f800-ee45-11e8-9dca-cfa7bef88277.png)

I confirm that increasing INFLUXDB_DATA_MAX_SERIES_PER_DATABASE to 1500000 has not solved the problem described in my ticket.

Note that the influxdb "partial write" error could be found in my logs but only a bunch of them every 3:30 hours, however, my graphs are cut every 5 minutes.

Please have a look at the 2 attached log files.
[http_codes_trace.txt](https://github.com/ntop/ntopng/files/2607447/http_codes_trace.txt)
[grep_-n_partial_writes_syslog.txt](https://github.com/ntop/ntopng/files/2607448/grep_-n_partial_writes_syslog.txt)

**Comment by simonemainardi** [comment]
it seems that your influx didn't take the new setting. Indeed, the error still say 1000000

```
31:Nov 22 06:30:12 gatewayNG ntopng[25386]: {"error":"partial write: max-series-per-database limit exceeded: (1000000) dropped=6961"}
```

I would set it to zero for unlimited : https://github.com/influxdata/docs.influxdata.com/issues/710#issuecomment-249424327
Then restart influxdb

**Comment by clebig** [comment]
No, look at the date, that was before my changes and restart. I have no more error yet regarding partial writes...
`Nov 22 10:28:54 gatewayNG systemd[1]: Stopping InfluxDB is an open-source, distributed, time series database...

Nov 22 10:28:54 gatewayNG systemd[1]: Stopped InfluxDB is an open-source, distributed, time series database.

Nov 22 10:28:54 gatewayNG systemd[1]: Started InfluxDB is an open-source, distributed, time series database.
`

**Comment by clebig** [comment]
> it seems that your influx didn't take the new setting. Indeed, the error still say 1000000
> 
> ```
> 31:Nov 22 06:30:12 gatewayNG ntopng[25386]: {"error":"partial write: max-series-per-database limit exceeded: (1000000) dropped=6961"}
> ```
> 
> I would set it to zero for unlimited : [influxdata/docs.influxdata.com#710 (comment)](https://github.com/influxdata/docs.influxdata.com/issues/710#issuecomment-249424327)
> Then restart influxdb

I have just set this setting to 0 and restarted influxdb.

**Comment by clebig** [comment]
I still get 
`Nov 22 11:00:04 gatewayNG influxd[30875]: [httpd] ::1 - - [22/Nov/2018:11:00:04 +0100] "POST /write?db=ntopng HTTP/1.1" 400 91 "-" "-" 62677bc5-ee3d-11e8-801d-000000000000 199389
`
and this seems to be absolutely correlated with the cuts.

**Comment by clebig** [comment]
> actually 204 is a 'no content' error. @emanuele-f do you think there are cases where we can send empty files to ntopng?

I would think 204 means no content in return to the POST-ed data. So I would interpret it as no problem. 
400 as return to the POST-ed data however would mean that the data couldn't be recorded correctly.

**Comment by simonemainardi** [comment]
the influxdb log doesn't seem to be very informative regarding the 400 bad request. Can you use tcpdump to record the traffic that goes to influxdb and then open the capture with wireshark to see the actual POST-ed payload that caused the 400?

**Comment by clebig** [comment]
Yep, So 400 returns a JSON object with error: "partial write: max-series-per-database limit exceeded: (1000000) dropped=14989"

So it seems to confirm that influxdb didn't take into account the line "INFLUXDB_DATA_MAX_SERIES_PER_DATABASE=0" I have added at the bottom of /etc/influxdb/influxdb.conf and that my issue is related to this 1000000 limit.

[influxdb.pcap.gz](https://github.com/ntop/ntopng/files/2607580/influxdb.pcap.gz)

**Comment by clebig** [comment]
I was setting it at the very bottom on the conf file when it was supposed to be set in [data]...
Now it's set in the data section. let see !

**Comment by clebig** [comment]
So... The setting is not taken into account.

When I do "influxd config" I can see the default limit (1000000) in the [data] section. I'm trying to understand why. So not a ntopng related problem.

**Comment by clebig** [comment]
Ok, the correct setting to put under [data] section is : 
`max-series-per-database = 0`
That works. I'll wait a little bit for the data to flow in and see if the problem is definitly solved.

**Comment by simonemainardi** [comment]
Thanks for reporting. I've already opened an issue to better handling errors, including this: https://github.com/ntop/ntopng/issues/2170

**Comment by clebig** [comment]
I confirm my problem is solved ! Thank you all.

**Comment by lucaderi** [comment]
See also https://github.com/influxdata/influxdb/issues/8432

**Comment by lucaderi** [comment]
So it seems from the ticket above that 204 actually means ok, so we're good here

**Comment by clebig** [comment]
Normal HTTP POST writes (with http return code 204) logs shouldn't be flowing into the OS kernel logging subsystem in a production environment. 
In the current realease of influxdb (1.7.1), an option is available to filter the information to be added to the logs based on the http return code.
The related option is "access-log-status-filter", [the documentation](https://github.com/influxdata/influxdb/blob/master/etc/config.sample.toml) on the [master branch of influxdb](https://github.com/influxdata/influxdb) project says :

\# Filters which requests should be logged. Each filter is of the pattern NNN, NNX, or NXX where N is 
\# a number and X is a wildcard for any number. To filter all 5xx responses, use the string 5xx.
\# If multiple filters are used, then only one has to match. The default is to have no filters which
\# will cause every request to be printed.
access-log-status-filters = []

So adding 
`access-log-status-filter = ["5xx", "4xx"]` in the [http] section would do the job.

But sadly... There is a bug affecting the current version (1.7.1) for which [a pull request for the fix has been approved for 1.7.2 but has not been yet merged/backported to 1.7.1](https://github.com/influxdata/influxdb/pull/10494/commits)

However, as a temporary measure until a patched version of influxdb is packaged, the http log entries can be deactivated using 
`suppress-write-log = true` in the [http] section of influxdb.conf
